Source/Details/_ASDisplayView.mm

```diff
@@ -358,14 +359,79 @@ - (void)tintColorDidChange
   [node tintColorDidChange];
 }
 
-- (BOOL)canBecomeFirstResponder {
-  ASDisplayNode *node = _asyncdisplaykit_node; // Create strong reference to weak ivar.
-  return [node canBecomeFirstResponder];
+#pragma mark UIResponder Handling
+
+#define HANDLE_RESPONDER_METHOD(__sel) \
+  ASDisplayNode *node = _asyncdisplaykit_node; \
+  SEL sel = @selector(__sel); \
+  /* Prevent an infinite loop in here if [super canBecomeFirstResponder] was called on a
+  / _ASDisplayView subclass */ \
+  if (ASSubclassOverridesSelector([_ASDisplayView class], [self class], sel)) { \
+    /* Check if we can call through to ASDisplayNode subclass directly */ \
+    if (ASDisplayNodeSubclassOverridesSelector([node class], sel)) { \
+      return [node __sel]; \
+    } else { \
+    /* Call through to views superclass as we expect super was called from the
+      _ASDisplayView subclass and a node subclass does not overwrite canBecomeFirstResponder */ \
+      return [self __##__sel]; \
+    } \
+  } else { \
+    /* Call through to internal node __canBecomeFirstResponder that will consider the view in responding */ \
+    return [node __##__sel]; \
+  } \
+
+
+- (BOOL)canBecomeFirstResponder
+{
+  HANDLE_RESPONDER_METHOD(canBecomeFirstResponder);
 }
 
-- (BOOL)canResignFirstResponder {
-  ASDisplayNode *node = _asyncdisplaykit_node; // Create strong reference to weak ivar.
-  return [node canResignFirstResponder];
+- (BOOL)becomeFirstResponder
+{
+  HANDLE_RESPONDER_METHOD(becomeFirstResponder);
+}
+
+- (BOOL)canResignFirstResponder
+{
+  HANDLE_RESPONDER_METHOD(canResignFirstResponder);
+}
+
+- (BOOL)resignFirstResponder
+{
+  HANDLE_RESPONDER_METHOD(resignFirstResponder);
+}
+
+- (BOOL)isFirstResponder
+{
+  HANDLE_RESPONDER_METHOD(isFirstResponder);
+}
+
+// This methods are called from ASDisplayNode to let the view decide in what UIResponder state they are not overwritten
+// by a ASDisplayNode subclass
+
+- (BOOL)__canBecomeFirstResponder
```

Comment:
Technically these could be macro-ized as well, generated in a pair with the above ones :). Optional of course.

Nits: 2 spaces instead of 4 on the returns. Also, "overridden" instead of "overwritten" in the comment.

Reply:
Also moved this into the macro and to 2 spaces.